File: solutions/src/tools/tools.cpp
#include "tools.hpp"

using namespace std;

tools::tools():a(0),b(0)
{
  cout << "Appel du constructeur par defaut de tools." << endl;
}

bool tools::set_tools(int val_a, int val_b)
{
  a = val_a; b = val_b;
  cout << "appel de set_tools." << endl;
  return true;
}

void tools::affiche() const
{
  cout << "Tools: a = "<< a << ", b = "<< b << endl;
}

File: solutions/src/tools/tools.f90
!> Description du module 
!!
!! on peut aussi mettre du latex: 
!! 	\f[
!!	g(\dot x, x, t, z) = 0
!!	\f]
module tools
implicit none

!> Une variable globale
integer, parameter :: var = 8

contains
  !> Description de test fortran
  !! \param val un entier
  subroutine affiche(val)
    real(kind=8) :: val 
    print *, "Appel de la routine affiche en fortran"
  end subroutine affiche
end module tools
